Lib/test/test_zlib.py

```diff
@@ -751,7 +751,12 @@ def test_large_unconsumed_tail(self, size):
     def test_wbits(self):
         # wbits=0 only supported since zlib v1.2.3.5
         # Register "1.2.3" as "1.2.3.0"
-        v = (zlib.ZLIB_RUNTIME_VERSION + ".0").split(".", 4)
+        if zlib.ZLIB_RUNTIME_VERSION.find('-')>=0:
+            #  zlib can report "1.2.8-linuxfoundation-mods-v1 or "1.2.8.f-linuxfoundation-mods-v1"
+            v = list(map( int , zlib.ZLIB_RUNTIME_VERSION.split('-',1)[0].split('.')[:3] ))
+            v.append(0)
+        else:
+            v = (zlib.ZLIB_RUNTIME_VERSION + ".0").split(".", 4)
```

Comment:
You should factorize the code between the if/else block. Something like:

v = zlib.ZLIB_RUNTIME_VERSION
if "-" in v:
  v = v.split('-',1)[0]
v = (v + ".0").split(".", 4)